### DISCUSSION TITLE
Support for other distros

### DISCUSSION THREAD

**Discussion by philipp-tty** [pivpn/pivpn#1605] [Feature requests]
Hello, I would love for the install script to support other distros like fedora. I saw an issue from 2017 that mentioned CentOS support but as far as I am aware the script checks for Raspian, Debian, Ubuntu and Alpine.

You can manually update/change the package manager but I am not sure if changing it to dnf and patching out the os support function would make the script work on fedora.

I don't have a testing installation right now and don't wanna try this out on my production mashine.

An official docker image would be a sufficient solution as well. The most popular one hasn't received an update in 4 years sadly. https://hub.docker.com/r/ljishen/pivpn

**Comment by coolapso** [comment]
Hi, thanks for your suggestion, for now we have no intention of supporting other distros, there is a bit more than just "change the package manager", most likely some packages will have different names, and quite possibly different config files locations. 

In regards to docker, we are slowly moving towards it, and that was the  main reason why we agreed with adding support for alpine linux. 

I strongly discourage using any pivpn docker image, specially that one, it raises some serious security concerns. everyone using that image is sharing the same encryption keys!

**Reply by philipp-tty** [reply]
Thank you for the reply. I understand that supporting more distributions means a lot of work.

Until we have an official image maybe we could create a guide for creating your own docker image. So the team wouldn't have to support it themselves but interested users can create their own image.

I am personally not sure on what to pay attention for creating an image myself.

**Reply by coolapso** [reply]
With alpine support it is now also possible to run it inside docker containers, however its very experimental and it has not been tested a lot, but here are some pointers: https://docs.pivpn.io/install/#docker-experimental

**Reply by coolapso** [reply]
I am planning to provide a Dockerfile here at some point.

**Reply by philipp-tty** [reply]
Well I guess you could just use an ubuntu base image but alpine support is a big plus. PiVPN has such a good user experience it would be a shame limiting it only to pi's and debian based x64 machines.
I well maintained and well documented image would probably become the go to choice for selfhosters and small to medium sized applications. 

Thank you (and all other contributors) for the time and effort you put into this project. ❤️
